rest-api-cloud-client/src/main/java/com/stnetix/ariaddna/externalcloudapi/cloudinterface/iAbstractCloud.java

```diff
@@ -0,0 +1,100 @@
+package com.stnetix.ariaddna.externalcloudapi.cloudinterface;
+
+/**
+ * This is an abstract interface containing general methods for
+ * manipulation with resources at any external cloud storage.
+ * Every certain cloud storage realization must implement this interface.
+ */
+public interface iAbstractCloud {
+    /**
+     * An uploadResource method implementation takes as a parameter a path to a resource
+     * which is stored at user's local HDD/SSD and sends it to an external cloud storage.
+     * Notice, that resource may be a file or directory with files.
+     *
+     * @param path A path to a resource at user's local storage.
+     * @return String representation of response with usefull
+     * information which may be converted into JSON Object.
+     */
+
+    String uploadResource(String path);
+
+    /**
+     * Provides the ability to upload a resource from one cloud disk storage to another.
+     * A method should somehow request a URL for downloading and pass it as parameter.
+     * Also a URL may point to any file that is available at the Internet.
+     * @param path A path where the uploaded resource should be stored.
+     * @param url A URL to an external resource.
+     * @return String representation of response with usefull
+     * information which may be converted into JSON Object.
+     */
+    String uploadExternalResource(String path, String url);
```

Comment:
And I think that we should use some object as return value. We can create abstract class JsonObject (if  we have dependencies from JSON view) and child classes ExternalResource.